File: framework/java/src/main/java/soda/unittest/job/codec/IntArray2dCodec.java
package soda.unittest.job.codec;

import java.util.List;

import soda.unittest.DataUtils;

public class IntArray2dCodec implements ICodec<List<List<Integer>>, int[][]> {

	@Override
	public List<List<Integer>> encode(int[][] object) {
		return DataUtils.toList2d(object);
	}

	@Override
	public int[][] decode(List<List<Integer>> serial) {
		return DataUtils.toIntArray2d((List<?>) serial);
	}

}

File: framework/scala/src/main/scala/soda/scala/unittest/codec/IntArray2dCodec.scala
package soda.scala.unittest.codec

import java.util.List

import soda.unittest.DataUtils

class IntArray2dCodec extends ICodec[List[List[Integer]], Array[Array[Int]]] {

    override def encode(obj: Array[Array[Int]]): List[List[Integer]] = {
        DataUtils.toList2d(obj)
    }

    override def decode(ser: List[List[Integer]]): Array[Array[Int]] = {
        DataUtils.toIntArray2d(ser)
    }

}

